_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,933 @@
+- Tag: **latest**
+
+Successful execution will show all stages in a green state:
+
+{% img blog/jhipster-ci-cd/sp-pipeline-success.png alt:"Spinnaker pipeline successful execution" width:"800" %}{: .center-image }
+
+
+Get the external IP of the gateway:
+
+```shell
+kubectl get svc gateway -n demo
+```
+
+Update the redirect URIs in Okta to allow the gateway address as a valid redirect. Run `okta login`, open the returned URL in your browser, and sign in to the Okta Admin Console. Go to the **Applications** section, find your application, edit, and add:
+- Sign-in redirect URIs: `http://<external-ip>:8080/login/oauth2/code/oidc`
+- Sign-out redirect URIs: `http://<external-ip>:8080`
+
+Navigate to `http://<external-ip>:8080` and sign in to the application with your Okta credentials. Then try creating a `Product`:
+
+{% img blog/jhipster-ci-cd/app-create-product.png alt:"Create a product in the store application" width:"800" %}{: .center-image }
+
+### Trigger the CI-CD pipeline with a GitHub push
+
+For testing the workflow, make a code change in the gateway. Edit `src/main/webapp/content/scss/_bootstrap-variables.scss` and update the following variable:
+
+```scss
+$body-bg: steelblue;
+```
+Also, the pipeline under test will only trigger if a new image tag is detected. So edit `.circleci/confg.yml` and update the image name:
+
+```yml
+IMAGE_NAME: indiepopart/gateway:v1
+```
+
+Commit and push the change to the `main` branch, and watch the CircleCI CI pipeline triggers. After the new gateway image is pushed to DockerHub, watch the Spinnaker CD pipeline trigger and deploy the updated gateway. On the left menu choose **Clusters**, and verify the active gateway deployment now has a _V002_. If you click over the **V002** box, you can also verify the image tag that was deployed.
+
+{% img blog/jhipster-ci-cd/sp-version-2.png alt:"Spinnaker deployment version 2" width:"800" %}{: .center-image }
+
+### Inspect Spinnaker logs
+
+For pipeline and trigger debugging, the Spinnaker services `spin-echo` and `spin-igor` inform Docker monitoring events and indicate the reasons why an execution was skipped.
+
+```shell
+kubectl get pods -n spinnaker
+kubectl logs spin-igor-7c8bdd94f5-lx5dl -n spinnaker | grep v1
+```
+```text
+2022-08-05 03:20:27.285  INFO 1 --- [   scheduling-1] c.n.spinnaker.igor.docker.DockerMonitor  : Found 1 new images for docker-account. Images: [{imageId=igor:dockerRegistry:v2:docker-account:indiepopart/gateway:v1, sendEvent=true}]
+2022-08-05 03:20:27.286  INFO 1 --- [   scheduling-1] c.n.spinnaker.igor.docker.DockerMonitor  : New tagged image: account=docker-account, image=igor:dockerRegistry:v2:docker-account:indiepopart/gateway:v1. Digest is now [null].
+2022-08-05 03:20:27.288  INFO 1 --- [   scheduling-1] c.n.spinnaker.igor.docker.DockerMonitor  : Sending tagged image info to echo: account=docker-account: image=igor:dockerRegistry:v2:docker-account:indiepopart/gateway:v1
+```
+
+## Spinnaker features and best practices  
```

Comment:
This section kinda seems like a wall of text that might be unnecessary. What do you think @havi?

Reply:
I updated the logs, I think it is important to see the logs when a new image is detected. I don't know why the horizontal scroll bars don't work for me.